src/Extend/Theme.php

```diff
@@ -52,6 +53,53 @@ public function overrideFileSource(string $file, string $newFilePath, string $ex
         return $this;
     }
 
+    /**
+     * This method allows you to add custom Less functions.
+     *
+     * **Example usage:**
+     * ```php
+     * (new Extend\Theme)
+     *     ->addCustomLessFunction('is-flarum', function (mixed $text) {
+     *         return new Less_Tree_Quoted('', strtolower($text) === 'flarum' ? 'true' : 'false')
+     *     }),
+     * ```
+     *
+     * @see https://leafo.net/lessphp/docs/#custom_functions
+     *
+     * @param string $functionName Name of the function identifier.
+     * @param callable $callable The PHP function to run when the Less function is called.
+     * @return self
+     */
+    public function addCustomLessFunction(string $functionName, callable $callable): self
+    {
+        if (! is_callable($callable)) {
+            return $this;
+        }
+
+        $this->customFunctions[$functionName] = function (...$args) use ($callable) {
+            $argVals = array_map(function ($arg) {
+                return $arg->value;
+            }, $args);
+
+            $return = $callable(...$argVals);
+
+            // Numbers
+            if (! is_string($return) && is_numeric($return)) {
+                return new \Less_Tree_Dimension($return);
+            }
+
+            // Strings
+            if (is_string($return)) {
+                return new \Less_Tree_Quoted('', $return);
+            }
+
+            // Anything else
+            return new \Less_Tree_Quoted('', (string) $return);
```

Comment:
What if the function returns `Less_Tree_Quoted already, as per the example in the docblock?

Reply:
They shouldn't return Less tree items. Docblock should have been updated.